src/framework/uicomponents/qml/MuseScore/UiComponents/StyledMenu.qml

```diff
@@ -21,11 +21,45 @@ StyledPopupView {
     property alias model: view.model
     property int itemWidth: 300
     property bool reserveSpaceForInvisibleItems: true
+    property bool unifyCheckMarksAndIcons: true
 
     signal handleAction(string actionCode, int actionIndex)
 
+    onModelChanged: {
+        privateProperties.hasItemsWithIconAndCheckable = false
+        privateProperties.hasItemsWithIconOrCheckable = false
+        privateProperties.hasItemsWithSubmenu = false
+        privateProperties.hasItemsWithShortcut = false
+
+        for (let i = 0; i < model.length; i++) {
+            let modelData = model[i]
+            if (!privateProperties.hasItemsWithIconAndCheckable
+                    && modelData.checkable
+                    && (Boolean(modelData.icon) && modelData.icon !== IconCode.NONE)) {
+                privateProperties.hasItemsWithIconAndCheckable = true
+                privateProperties.hasItemsWithIconOrCheckable = true
+            } else if (!privateProperties.hasItemsWithIconOrCheckable
+                       && (modelData.checkable || (modelData.icon && modelData.icon !== IconCode.NONE))) {
```

Comment:
You have this code in several places:
modelData.icon && modelData.icon !== IconCode.NONE

It is better to add some temp variable for this condition after this line
let modelData = model[i]